lib/status/status.go

```diff
@@ -357,7 +362,31 @@ func probeErrorDetail(p pb.Probe) string {
 	if p.Detail == "" {
 		detail = p.Checker
 	}
-	return fmt.Sprintf("%v failed", detail)
+	// state directory disk space checker always checks /var/lib/gravity
+	// which is default path inside planet but may be different on host
+	// so determine the real state directory if needed
+	if strings.HasPrefix(p.Checker, monitoring.DiskSpaceCheckerID) {
```

Comment:
Please see my comment about embedding path into the checker name and the resulting need to parse it here.
Also, it looks like `probeFromErr` has blown up and turned into something hard to read - can you factor this new code out?
On the other hand, this code looks completely out of place here, inside a simple `probeErrorDetail` which used to extract the detail information from a probe - this should be done somewhere up the call chain.

Reply:
It can't be done up the chain because the "Probe" object (along with its details) is not available there anymore. This is the place the object basically gets converted to a string so I think this is a suitable place to construct a proper error message. I can factor this logic out though so it does not look so intimidating.